image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
 
-=== Use Services for inter-image communication
-For cases where your image needs to communicate with a service provided by another image (e.g. a web frontend image that needs to access a database image to store/retrieve data), your image should consume an OpenShift Service.  Services provide a static endpoint for access which will not change as containers are stopped/started/moved.  In addition services provide load balancing for requests.  For more information see https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/services.md[this documentation].  (NOTE to docs team:  this link should really go to something in the openshift docs once we have it)
+It is also possible and recommended to pass secrets such as certificates and keys into the container using environment variables. This ensures that the secret values do not end up committed in an image and leaked into a Docker registry.
 
-=== Provide common libraries
-For images that are intended to run application code provided by a third party, ensure that your image contains commonly used libraries for your platform.  In particular, provide database drivers for common databases used with your platform (e.g. provide JDBC drivers for MySQL and PostgreSQL if you are creating a java framework image).  This will prevent the need for common dependencies to be downloaded during application assembly time, speeding up application image builds.  It will also simplify the work application authors need to do to ensure all their dependencies are met.
+Providing environment variables allows consumers of your image to customize behavior, such as database settings, passwords, and performance tuning, without having to introduce a new layer on top of your image. Instead, they can simply define environment variable values when defining a pod and change those settings without rebuilding the image.
 
-=== Configuration
-Users of your image should be able to configure it without having to create a downstream image based on your image.  This means that runtime configuration should be handled via environment variables.  For simple configuration, the running process can consume the environment variables directly.  For more complicated configuration or for runtimes which do not support this, you should configure the runtime by defining a template configuration file which is processed during startup.  During this processing, values supplied via environment variable can be substituted into the configuration file or used to make decisions about what options to set in the configuration file.
+For extremely complex scenarios, configuration can also be supplied using volumes which are expected to be mounted into the container at runtime. If you take this path, your image should provide clear error messages on startup when the necessary volume or configuration is not present.
 
```

Comment:
[Suggestion]
For extremely complex scenarios, configuration can also be supplied using volumes that would be mounted into the container at runtime. However, if you elect to do it this way you must ensure that your image provides clear error messages on startup when the necessary volume or configuration is not present.